core/src/main/resources/db/changelog/changesets/2024_02_16_language_texts.yaml

```diff
@@ -0,0 +1,18 @@
+databaseChangeLog:
+  - changeSet:
+      id: add_core_language_text
+      author: mbemis
+      changes:
+        - createTable:
+            tableName: core_language_text
+            columns:
+              - column: { name: id, type: uuid, defaultValueComputed: gen_random_uuid(), constraints: { nullable: false, primaryKey: true } }
+              - column: { name: created_at, type: datetime, constraints: { nullable: false } }
+              - column: { name: last_updated_at, type: datetime, constraints: { nullable: false } }
+              - column: { name: i18n_key, type: text, constraints: { nullable: false } }
+              - column: { name: text, type: text, constraints: { nullable: false } }
+              - column: { name: language, type: text, constraints: { nullable: false } }
```

Comment:
by far our most common case is going to be pulling all the messages out for a given language. I think it would be worth 5mins seeing if there's a postgresql index type we can put on the `language` column that would make that retrieval faster.  I think a regular index won't help, since IIRC postgresql ignores indexes if it suspects the query will return more than ~1% of the total rows.  But there might be some sort of partition/sorting index that will force postgres to store all the messages in a given language together (at the cost of higher insert time, but we don't care about that)

Reply:
Yeah, the cardinality of `language` kind of makes a lot of the index options worthless. I looked a bit to see if there was a way to get Postgres to cluster them together and didn't find anything that looked appealing at the moment. But if we hit a performance wall in the future, we could maybe use the `CLUSTER` command on this table every time the application-related texts repopulate on boot to keep entries for each language near each other. I don't know if that's crazy, my DB skills are rusty. But there's I imagine there's something we can do with more research.

Hopefully the Spring-level caching alleviates some of the performance concerns in the meantime.